pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,44 @@
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+
+// TrainedModelList contains a list of TrainedModel
+// +kubebuilder:object:root=true
+type TrainedModelList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// +listType=set
+	Items []TrainedModel `json:"items"`
+}
+
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
```

Comment:
model.predictorModel stutters. Predictor doesn't really make sense, because the predictor is the architectural component in KFServing. I see this model as more of a data definition than a thing-do-er definition. How about model.spec.inference.storageUri and model.spec.explaination.storageUri?

Reply:
How about just rename it to ModelDefinition? I'd like to keep this struct since Puller can also use it.

Reply:
I renamed PredictorModel to Inference. I think we need more discussion before we add the explanation field.

Reply:
I only said explainer as to illustrate how we might extend this in the future. Let's punt for now :)